src/engine/image_tool.cpp

```diff
@@ -325,6 +325,24 @@ namespace fheroes2
         return sprite;
     }
 
+    void decodeTILImages( const uint8_t * data, const size_t imageCount, const int32_t width, const int32_t height, std::vector<Image> & output )
+    {
+        assert( data != nullptr && imageCount > 0 && width > 0 && height > 0 );
+
+        output.clear();
+        output.resize( imageCount );
+
+        const size_t imageSize = static_cast<size_t>( width * height );
```

Comment:
We can avoid this warning this way:

```suggestion
        const size_t imageSize = static_cast<size_t>( width ) * height;
```